pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java

```diff
@@ -259,13 +295,34 @@ private boolean verifyAuthenticationIfNeeded(CommandConnect connect) {
             clientAuthRole = service.getAuthenticationService().authenticate(authenticationData, authMethod);
             LOG.info("[{}] Client successfully authenticated with {} role {}", remoteAddress, authMethod,
                     clientAuthRole);
+            if (service.getConfiguration().forwardAuthorizationCredentials()) {
+                this.clientAuthData = authData;
+                this.clientAuthMethod = authMethod;
+            }
+            createClient(clientConf, this.clientAuthData, this.clientAuthMethod);
```

Comment:
as name suggests. can we create and return client in `createClient()`: `this.client = createClient(..)`

Reply:
also we don't want to share proxy-to-broker connection when `service.getConfiguration().forwardAuthorizationCredentials()` is enabled. else, we can always share proxy-to-broker connection for lookup. 
so, should we create a new client only in case of `service.getConfiguration().forwardAuthorizationCredentials()=true` else we can always share it.?